akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala

```diff
@@ -289,20 +289,8 @@ object Source {
    *   }
    * }}}
    */
-  def unfoldInf[S, E](s: S)(f: S ⇒ (S, E)): Source[E, Unit] = {
-    Source.fromGraph(GraphDSL.create() { implicit b ⇒
-      import GraphDSL.Implicits._
-
-      val uzip = b.add(UnzipWith(f))
-      val cnct = b.add(Concat[S]())
-      val init = Source.single(s)
-
-      init ~> cnct ~> uzip.in
-      cnct <~ Flow[S].buffer(2, OverflowStrategy.backpressure) <~ uzip.out0
-
-      SourceShape(uzip.out1)
-    }).withAttributes(DefaultAttributes.unfoldInf)
-  }
+  def unfoldInf[S, E](s: S)(f: S ⇒ (S, E)): Source[E, Unit] =
+    unfold(s)(s ⇒ Some(f(s))).withAttributes(DefaultAttributes.unfoldInf)
```

Comment:
@drewhk When I remove `unfoldInf` and relace the usage thereof in `FusingSpec` it fails, because it is introspecting the structure of the graph, and the name-attribute of `Unfold` gets hidden in a CopiedModule where it's `attributes` do not match `copyOf.attributes`, so if I do `unfold(…).named("unfold").named("unfold")` the test passes.

I'm not sure whether the FusingSpec `def singlePath` makes improper assumptions or whether `CopiedModuled` has a bug. thoughts?